Std/Data/Bool.lean

```diff
@@ -0,0 +1,166 @@
+alias Bool.false_xor ← xor_false_left
+alias Bool.xor_false ← xor_false_right
+alias Bool.true_xor ← xor_true_left
+alias Bool.xor_true ← xor_true_right
+
+@[simp] protected theorem le_refl : x ≤ x := by btt x
+@[simp] protected theorem lt_irrefl : ¬ x < x := by btt using simp x
+protected theorem le_trans {x y z : Bool} : x ≤ y → y ≤ z → x ≤ z := by btt using simp x y z
+protected theorem le_antisymm {x y : Bool} : x ≤ y → y ≤ x → x = y := by btt using simp x y
+protected theorem le_total (x y : Bool) : x ≤ y ∨ y ≤ x := by btt using simp x y
+protected theorem lt_asymm {x y : Bool} : x < y → ¬ y < x := by btt using simp x y
+protected theorem lt_trans {x y z : Bool} : x < y → y < z → x < z := by btt using simp x y z
+protected theorem lt_iff_le_not_le {x y : Bool} : x < y ↔ x ≤ y ∧ ¬ y ≤ x := by btt using simp x y
+protected theorem lt_of_le_of_lt {x y z : Bool} : x ≤ y → y < z → x < z := by btt using simp x y z
+protected theorem lt_of_lt_of_le {x y z : Bool} : x < y → y ≤ z → x < z := by btt using simp x y z
+protected theorem le_of_lt {x y : Bool} : x < y → x ≤ y := by btt using simp x y z
+protected theorem le_of_eq {x y : Bool} : x = y → x ≤ y := by btt using simp x y z
+protected theorem ne_of_lt {x y : Bool} : x < y → x ≠ y := by btt using simp x y z
+protected theorem lt_of_le_of_ne {x y : Bool} : x ≤ y → x ≠ y → x < y := by btt using simp x y z
+protected theorem le_of_lt_or_eq {x y : Bool} : x < y ∨ x = y → x ≤ y := by btt using simp x y z
+@[simp] protected theorem le_true : x ≤ true := by btt x
+@[simp] protected theorem false_le : false ≤ x := by btt x
+protected theorem eq_true_of_true_le {x : Bool} : true ≤ x → x = true := by btt using simp x
+protected theorem eq_false_of_le_false {x : Bool} : x ≤ false → x = false := by btt using simp x
+
+protected theorem max_eq_or (x y : Bool) : max x y = (x || y) := by btt using simp x y
+protected theorem min_eq_and (x y : Bool) : min x y = (x && y) := by btt using simp x y
+
+theorem not_inj {x y : Bool} : (!x) = (!y) → x = y := by btt using simp x y
+
+theorem not_inj' {x y : Bool} : (!x) = (!y) ↔ x = y := ⟨not_inj, fun | rfl => rfl⟩
+
+theorem and_or_inj_right {m x y : Bool}: (x && m) = (y && m) → (x || m) = (y || m) → x = y := by
+  btt using simp m x y
+
+theorem and_or_inj_right' {m x y : Bool}: (x && m) = (y && m) ∧ (x || m) = (y || m) ↔ x = y :=
+  ⟨fun ⟨h₁, h₂⟩ => and_or_inj_right h₁ h₂, fun | rfl => ⟨rfl, rfl⟩⟩
+
+theorem and_or_inj_left {m x y : Bool} : (m && x) = (m && y) → (m || x) = (m || y) → x = y := by
+  btt using simp m x y
+
+theorem and_or_inj_left' {m x y : Bool} : (m && x) = (m && y) ∧ (m || x) = (m || y) ↔ x = y :=
+  ⟨fun ⟨h₁, h₂⟩ => and_or_inj_left h₁ h₂, fun | rfl => ⟨rfl, rfl⟩⟩
+
+section
+variable (xs ys : List Bool)
+
+attribute [local simp] and_assoc or_assoc xor_assoc List.map List.join
+
+/-- Boolean conjunction (`and`) of a list -/
+abbrev all : Bool := xs.all id
+
+@[simp] theorem all_nil : all [] = true := rfl
```

Comment:
Can't `simp` prove this using `simp` lemmas for `List.all`?
Do we need an abbreviation for a special case of `List.all`?

Reply:
That was a remnant from when I was defining `List.all` as `Bool.all (List.map _ _)` in my personal libraries. It's no longer needed.